yesod-core/src/Yesod/Core/Class/Yesod.hs

```diff
@@ -202,6 +202,18 @@ class RenderRoute site => Yesod site where
     maximumContentLength :: site -> Maybe (Route site) -> Maybe Word64
     maximumContentLength _ _ = Just $ 2 * 1024 * 1024 -- 2 megabytes
 
+    -- | Maximum allowed length of the request body, in bytes. This is similar
+    -- to 'maximumContentLength', but the result lives in @IO@. This allows
+    -- you to dynamically change the maximum file size based on some external
+    -- source like a database or an @IORef@.
+    --
+    -- The default implementation uses 'maximumContentLength'. Future version of yesod will
```

Comment:
How about adding a comment to `maximumContentLength` as well that overriding this method will cause `maximumContentLength` to be ignored?

Reply:
I've pushed f1374c91409f2e22a0b1cc8890e87fbbd8d7d880 but I'm not sure this is what you wanted.